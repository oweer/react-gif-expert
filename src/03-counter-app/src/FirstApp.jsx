import PropTypes from 'prop-types';

export const FirstApp = ({ 
  name = 'Belly', 
  title = 'No hay título', 
  subTitle = 'No hay subtítulo' 
}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
      <p>{name}</p>
      
    </>
  );
};

