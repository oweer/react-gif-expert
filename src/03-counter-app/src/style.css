html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 10px;
}
button{
    padding: 5px;
    font-size: 1.2rem;
    margin-right: 5px;
}