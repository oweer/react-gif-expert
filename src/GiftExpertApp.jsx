import { useState } from "react";
import { AddCategory, GiftGrid } from "./components";


export const GiftExpertApp = () => {

    const [categories, setCategories] = useState([ 'Twilight']);

    const onAddCategory= ( newCategory )=>{

      if( categories.includes(newCategory)) return;
      setCategories( cat => [newCategory,...categories]);
      
    };

  return (
    <> 
        <h1>GiftExpertApp</h1>

      <AddCategory 
      onNewCategory={ (event) => onAddCategory(event)}
      />

          {categories.map( category => 
                (
                <GiftGrid 
                key={category} 
                category = {category}/>
            ))
          } 


    </>
  )
}
