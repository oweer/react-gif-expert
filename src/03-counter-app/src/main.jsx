import React from 'react';
import ReactDOM from 'react-dom/client';
//import {HelloWorldApp} from './HelloWorldApp';
import {FirstApp} from './FirstApp';
import {CounterApp} from './CounterApp';
import './style.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
      {/* <HelloWorldApp /> */}
       <CounterApp value={100} />
      {/* <FirstApp title='Hola'/> */}
    </React.StrictMode>
);
