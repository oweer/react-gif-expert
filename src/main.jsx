import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { GiftExpertApp } from './GiftExpertApp';
import './style.css';


createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GiftExpertApp />
  </StrictMode>,
)
