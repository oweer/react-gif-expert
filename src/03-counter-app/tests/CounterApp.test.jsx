/**
 * @jest-environment jsdom
 */

import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas de <ConterApp/>", ()=>{

    const value = 10;

    test('Debe hacer match con el snapchot', ()=>{
        const {container} = render (<CounterApp value={value} />);
        expect( container).toMatchSnapshot();
    
      })

      test('Debe iniciar el 100', ()=>{
        const {container} = render (<CounterApp value={100} />);
        expect( screen.getByText(100)).toBeTruthy();

        // expect( screen.getByRole('heading',{level:2}).innerHTML).toContain('100')
    
      })

      test('Debe de incrementar con el boton +1', ()=>{

        render(<CounterApp value={value} />);
        fireEvent.click( screen.getByText('+1') )
        expect(screen.getByText('11')).toBeTruthy();

      })

      test('Debe de descrementar con el boton -1', ()=>{

        render(<CounterApp value={value} />);
        fireEvent.click( screen.getByText('-1') )
        expect(screen.getByText('9')).toBeTruthy();

      })

      test('Debe de resetar con el boton reset', ()=>{

        render(<CounterApp value={365} />);
        fireEvent.click( screen.getByText('-1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('-1') )
        // fireEvent.click( screen.getByText('reset') )

        fireEvent.click(screen.getByRole('button', {name:'btn-reset'}))
        
        expect(screen.getByText(365)).toBeTruthy();

      })

});