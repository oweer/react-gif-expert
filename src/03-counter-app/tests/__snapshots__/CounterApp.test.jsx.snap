// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <ConterApp/> Debe hacer match con el snapchot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
